src/Controller.js

```diff
@@ -80,20 +95,76 @@
     },
 
     renderMessage(message) {
-      const messageElement = document.createElement('div');
-      messageElement.id = 'message';
+      const messageElement = document.createElement("div");
+      messageElement.id = "message";
       messageElement.innerHTML = message;
 
-      const viewport = document.querySelector('viewport');
+      const viewport = document.querySelector("#viewport");
       viewport.appendChild(messageElement);
 
       setTimeout(() => {
         viewport.removeChild(messageElement);
       }, 2000);
     },
+
+    portUpdate() {
+      const ship = this.ship;
+
+      const currentPortElement = document.getElementById("current-port");
+      const nextPortElement = document.getElementById("next-port");
+
+      currentPortElement.textContent = `Current Port: ${ship.currentPort.name}`;
+
+      const currentPortIndex = ship.itinerary.ports.indexOf(ship.currentPort);
+      const nextPortIndex = currentPortIndex + 1;
+
+      if (nextPortIndex === ship.itinerary.ports.length) {
+        nextPortElement.textContent = "End of Cruise";
+      } else {
+        nextPortElement.textContent = `Next Port: ${ship.itinerary.ports[nextPortIndex].name}`;
+      }
+    },
+
+    typePort() {
+      const ship = this.ship;
+
+      const typedPort = document.getElementById("portName").value;
+      // console.log(typedPort);
+
+      if (typedPort === "") {
+        return this.renderMessage("Please type a port to be added!");
+      }
+
+      const expectedTypedPort = new Port(typedPort);
+      console.log(expectedTypedPort);
```

Comment:
It's fine to keep `console.log`s for the exercise, so you can monitor what you app does, but please be aware that these shouldn't go to production code.